### PR TITLE
Function overrides for `_handle`

### PR DESCRIPTION
**Pull request by EgeCaner**

### CODE REVIEW

File: cairo/src/contracts/token/components/fast_token_router.cairo [hunk 1]
```diff
@@ -57,6 +58,39 @@ pub mod FastTokenRouterComponent {
...
+            ref self: RouterComponent::ComponentState<TContractState>,
+            origin: u32,
+            sender: u256,
+            message: Bytes
+        ) {
+            let recipient = message.recipient();
+            let amount = message.amount();
+            let metadata = message.metadata();
+
+            let mut contract_state = RouterComponent::HasComponent::get_contract_mut(ref self);
```
[the remark this answers is not in the document]
Comment (EgeCaner) [reply cairo/src/contracts/token/components/fast_token_router.cairo:83]: self is `RouterComponent::ComponentState<TContractState>` here, I need to get the `FastTokenRouterComponent::ComponentState<TContractState>` for that I need to pass `TContractState`. Not sure can I derive it without having `TContract` state

[the remark this answers is not in the document]
Comment (EgeCaner) [reply cairo/src/contracts/token/components/fast_token_router.cairo:83]: I am getting this error when try to use self to switch to `FastTokenRouterComponent::ComponentState<TContractState>`

> error: Trait has no implementation in context: hyperlane_starknet::contracts::token::components::fast_token_router::FastTokenRouterComponent::HasComponent::<hyperlane_starknet::contracts::client::router_component::RouterComponent::ComponentState::<TContractState>>
 --> /Users/egecaner/dev/NM/hyperlane/hyperlane_starknet/cairo/src/contracts/token/components/fast_token_router.cairo:84:53
            let mut component_state = HasComponent::get_component_mut(ref self);

[the remark this answers is not in the document]
Comment (EgeCaner) [reply cairo/src/contracts/token/components/fast_token_router.cairo:83]: Or you meant something else?

[the remark this answers is not in the document]
Comment (ametel01) [reply cairo/src/contracts/token/components/fast_token_router.cairo:83]: Yes i was just wondering if that might have worked

**Pull request merged**